js/keyword-search.js

```diff
@@ -0,0 +1,212 @@
+namespacer('baltimoreCounty');
+
+baltimoreCounty.keywordSearch = (function keywordSearch($, sessionStorage, Handlebars) {
+	'use strict';
+
+	var searchData;
+
+	var documentClickHandler = function documentClickHandler() {
+		var $searchResults = $('#header-search-results');
+
+		if ($searchResults.is(':visible')) {
+			$searchResults.hide();
+		}
+	};
+
+	/**
+	 * Highlights the matched term in the results.
+	 * 
+	 * @param {String} searchTerm 
+	 * @param {Array<Term, Order>} matches 
+	 */
+	var highlightMatches = function highlightMatches(searchTerm, matches) {
+		var highlightedMatches = [];
+
+		matches.forEach(function forEachTopFiveMatch(match) {
+			var highlightedMatch = $.extend({}, match);
+			highlightedMatch.Term = highlightedMatch.Term.replace(searchTerm, '<strong>' + searchTerm + '</strong>');
+			highlightedMatches.push(highlightedMatch);
+		});
+
+		return highlightedMatches;
+	};
+
+	var init = function init(callback) {
+		if (sessionStorage && sessionStorage.searchData) {
+			searchData = JSON.parse(sessionStorage.searchData);
+		} else {
+			$.ajax('/sebin/m/m/searchTerms.json')
```

Comment:
move this to the top or a constants file.

Reply:
Moved to constants